app/frontend/js/ExtractionDefinitionFormLogic.js

```diff
@@ -33,6 +33,52 @@ if(extractionDefinitionPaginationTypeSelect) {
       element.style.display = 'none';
     });
   }
+
+  const extractionDefinitionAddHeader = document.getElementById('js-extraction-definition-add-header');
+
+  if(extractionDefinitionAddHeader) {
+    extractionDefinitionAddHeader.addEventListener('click', (event) => {
+
+      const formElements = document.querySelector('#js-extraction-definition-headers').getElementsByClassName('form-control');
+      const lastElement = formElements[formElements.length - 1];
+      let newId;
+
+      if(lastElement != undefined) {
+        const lastId = lastElement.id.match(/.+(?<id>\d)/).groups.id;
+        newId = parseInt(lastId) + 1;
+      } else {
+        newId = 0;
+      }
+
+      const headerForm = `
+        <div class='row gy-3 mb-5 align-items-center'>
+          <div class='col-4'>
+            <label class='form-label' for='extraction_definition_headers_attributes_${newId}_name'>
+              Header Name
+            </label>                
+          </div>
+          
+          <div class='col-8'>
+            <input class='form-control' type='text' name='extraction_definition[headers_attributes][${newId}][name]'' id='extraction_definition_headers_attributes_${newId}_name'>
+          </div>
+          
+          <div class='col-4'>
+            <label class='form-label' for='extraction_definition_headers_attributes_${newId}_value'>
+              Header Value
+            </label>                
+          </div>
+        
+          <div class='col-8'>
+            <input class='form-control' type='text' name='extraction_definition[headers_attributes][${newId}][value]'' id='extraction_definition_headers_attributes_${newId}_value'>
+          </div>
+        </div>
```

Comment:
Is it time to switch this form to react?

Reply:
Yes I think so. When we do the `PAGINATION ADJUSTMENTS` story I think that will be a good opportunity to replace this form with React.